### ISSUE TITLE
`mypy` does not report column and error end for `ignore-without-code`

### ISSUE THREAD

**Issue by still-wondering** [python/mypy#21856]
**Bug Report**

When calling `mypy` with `--show-column-numbers --show-error-end` the error end and column are reported.

**To Reproduce**

```python
# main.py
# mypy: enable-error-code="ignore-without-code"
a: int = "a" # type: ignore
```

```bash
mypy --show-column-numbers --show-error-end main.py
```

**Expected Behavior**

Expected the output to contain column and error end numbers:
```
main.py:3:14:3:27: error: "type: ignore" comment without error code (consider "type: ignore[assignment]" instead)  [ignore-without-code]
Found 1 error in 1 file (checked 1 source file)
```

The same way it is e.g. if the type ignore is removed in the example:
```
main.py:3:10:3:12: error: Incompatible types in assignment (expression has type "str", variable has type "int")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**

No column or error end is included, only the line number:
```
main.py:3: error: "type: ignore" comment without error code (consider "type: ignore[assignment]" instead)  [ignore-without-code]
Found 1 error in 1 file (checked 1 source file)
```

Calling `mypy` with only one of the flags results in the same behavior.

**Your Environment**

- Mypy version used: 2.3.0 (compiled)
- Mypy command-line flags: --show-column-numbers --show-error-end 
- Mypy configuration options from `mypy.ini` (and other config files): no configuration
- Python version used: 3.14.6 / 3.12.13

**Comment by Hmanbo** [comment]
I'd like to take a look at this and work on a fix. I'll start by reproducing the issue and investigating how the error range is reported.

**Comment by Hmanbo** [comment]
Thanks for the report — I reproduced this and found the cause. The `ignore-without-code` diagnostic is synthesized after type checking through `report_simple_error()`, which currently records both column fields as unknown.

I have a local fix that locates the actual `# type: ignore` comment token and passes its start and end columns to the diagnostic. With the reproducer, the output is now:

```text
main.py:3:14:3:27: error: "type: ignore" comment without error code (consider "type: ignore[assignment]" instead)  [ignore-without-code]
```

I also added a regression case that distinguishes an actual comment from the same text inside a string literal. The column/error-code suites (175 tests), the relevant daemon rerun test, and mypy's full self-check all pass, so this looks fixable with a focused patch.
